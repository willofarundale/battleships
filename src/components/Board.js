import React from "react";
import PropTypes from "prop-types";

import hit from '../icons/hit.svg';
import missed from '../icons/missed.svg';

const Board = ({ onClick, grid }) => {

  return (
    <div className="board">
      {grid.map((row, i) => (
        <div className="row" key={i}>
          {row.map((cell, i2) => (
            //onClick provides coordinates
            <div
              className="col"
              key={i2}
              onClick={() => onClick(`${i},${i2}`)}
            >
              {cell.hit === 0 && <span>_</span>}
              {cell.ship === 0 && cell.hit === 1 && <span className="missed"><img src={missed} alt="missed-icon"/></span>}
              {cell.ship === 1 && cell.hit === 1 && <span className="hit"><img src={hit} alt="hit-icon"/></span>}
            </div>
          ))}
        </div>
      ))}
    </div>
  );
};

//Set Default Props
Board.defaultProps = {
  grid: []
};

export default Board;
