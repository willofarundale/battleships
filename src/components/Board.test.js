import React from 'react';
import { shallow, mount} from 'enzyme';
import Board from './Board';

//Board Renders
describe('Board', () => {
  it('Renders', () => {
    const wrapper = shallow(<Board />);
  
    expect(wrapper.exists()).toBe(true);
  });
});


