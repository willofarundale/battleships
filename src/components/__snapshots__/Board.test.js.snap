// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Renders 1`] = `
<div
  className="board"
/>
`;
